### ISSUE TITLE
Color balance RGB: use module's masks for hue shift, vibrance and contrast 

### ISSUE THREAD

**Issue by pedrorrodriguez** [aurelienpierreeng/ansel#138]
Perhaps the masks are not smooth enough to do this, or you fear the ui would get too complex. However, if there is no technical limitation, there are a few arguments in favor of being able to have finer control for hue shift, vibrance and contrast, just like all other parameters in the module. 

Hue shift: it would eliminate the constant complaint that filmic gives sunsets a salmon look (it doesn't, but you could easily shift hl to your liking). It could be also used creatively to imitate the color shifts that can happen in film. The manual says it is best used masked too. 
EDIT: no, hue shifts in film don't quite work like this. The salmon thing can work, but the mask might need to be tweaked, and at that point, regular masking is more flexible. 

Vibrance: the most common use case imo would be increasing vibrance selectively while leaving highlights intact. 

Contrast: it would be useful to quickly and selectively tweak local contrast. It is no tone eq, but for finer adjustments it would be very useful. The manual also recommends it to be used masked. 

Against: perhaps masks are not smooth enough, ui complexity. 
For: control and speed, since the masks are already created by the module.

**Comment by aurelienpierre** [comment]
Hue is not an additive property but an angle, and as such it's incredibly complicated to mask-split it seamlessly. That's the realm of the color EQ and it's not super safe. The hue control in color balance is a despill feature, you shouldn't use it globally indeed, but it has no connection with input luminance.

Vibrance is mostly bullshit, it already selects low-chroma colors, I see no point in making it any more selective. Use saturation, at least it's a proper perceptual metric.

Contrast is by nature global, local contrast is related to image geometry (it's a convolutional filter), not a tone curve. Masking a tone curve will not have the local contrast effect you think it will have.

In conclusion, I think the GUI of this module has reached the maximum of complexity we could bare. For everything else, restrict the use with the typical masking features.

**Comment by pedrorrodriguez** [comment]
> Hue is not an additive property but an angle, and as such it's incredibly complicated to mask-split it seamlessly. That's the realm of the color EQ and it's not super safe. The hue control in color balance is a despill feature, you shouldn't use it globally indeed, but it has no connection with input luminance.

Understood 

> Vibrance is mostly bullshit, it already selects low-chroma colors, I see no point in making it any more selective. Use saturation, at least it's a proper perceptual metric.

Fine too. Not that useful anyways. 

> Contrast is by nature global, local contrast is related to image geometry (it's a convolutional filter), not a tone curve. Masking a tone curve will not have the local contrast effect you think it will have.

Didn't realize, but true. For it to increase contrast in the same way as globally you'd have to move the fulcrum, right? 

> In conclusion, I think the GUI of this module has reached the maximum of complexity we could bare. For everything else, restrict the use with the typical masking features.

Makes sense.
